src/ArmTemplates/Common/Extensions/NamingHelper.cs

```diff
@@ -5,13 +5,15 @@
 
 using System.Linq;
 using System.Text.RegularExpressions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions
 {
     public static class NamingHelper
     {
         static readonly Regex ExcludeOtherFromLettersAndDigitsRegex = new Regex("[^a-zA-Z0-9]");
         static readonly Regex ExcludeOtherFromAlphaNumericsAndHyphensRegex = new Regex("[^a-zA-Z0-9-]");
+        static readonly Regex ExcludeOtherFromValidDeploymentNameCharsRegex = new Regex(@"[^-\w\._\(\)]");
```

Comment:
I am saying about `ExcludeOtherFromValidDeploymentNameCharsRegex` to have an info about excluded characters. Because still what is `validDeploymentName` is a mystery.

You can leave name as it is, but provide a comment on how regex works.

Reply:
Description of the regex is added. Pretty big though

Reply:
this is awesome - thanks a ton for that!